docs/upgrade-guides/3-19-to-3-20.mdx

```diff
@@ -13,4 +13,151 @@ Otherwise, you will need to downtime your solution to ensure correct data migrat
\ No newline at end of file
+Please refer to [Amazon S3](../setup/media-s3) or [Google Cloud Storage (GCS)](../setup/media-gcs) guidelines if you are using cloud storage.
+
+## Automatic checkout completion
+
+Saleor 3.20 added new functionality to automatically complete `Checkout` that have been paid by the customer.
+
+👉 To learn more, about how this feature works and how to enable it, [read the Automatic Checkout completion docs](/developer/payments/transactions.mdx#automatic-checkout-completion).
+
+:::info
+By "paid" in this guide we mean that the `TransactionItems` connected to `Checkout` have fully covered [`Checkout.totalPrice`](/api-reference/checkout/objects/checkout.mdx#checkouttotalpricetaxedmoney---), which means that [`Checkout.authorizeStatus`](/api-reference/checkout/objects/checkout.mdx#checkoutauthorizestatuscheckoutauthorizestatusenum---) is set to [`FULL`](/api-reference/checkout/enums/checkout-authorize-status-enum.mdx).
+
+Learn more about this in [Transaction docs](/developer/payments/transactions.mdx#automatic-checkout-completion)
+:::
+
+This feature is **disabled** after migration, and needs to be enabled manually in the [channel settings](/developer/channels/configuration.mdx#settings).
+
+New Saleor instances created from version 3.20 onwards will have this feature **enabled by default**.
+
+This removes the need for storefront implementation to call `checkoutComplete` mutation in order to create `Order`.
+
+Previously a store might have had missing orders, due to `checkoutComplete` mutation not being always called. This could have happened for many unexpected reasons: network issues, storefront bug, customer leaving website, closing tab before being redirected from payment provider, etc.
+
+:::tip
+Before Saleor 3.20 you might have used other measures to mitigate missing orders issue, for example by using `CHECKOUT_FULLY_PAID` webhook and calling `checkoutComplete` mutation on your backend.
+
+After migrating to Saleor 3.20 and enabling this feature for all your channels, you can safely get rid of this workaround.
+
+Keep in mind that `CHECKOUT_FULLY_PAID` webhook would have only been triggered if you also used `TransactionFlowStrategy.CHARGE` for your channel. Automatic checkout completion doesn't have this problem.
+:::
+
+
+### Before you enable this feature
+
+
+:::danger
+Enabling this feature might break your current storefront implementation and services connected to Saleor that use `Checkout` object.
+:::
+
+Before you enable this feature make sure that you don't rely on any `Checkout` query or mutation that updates `Checkout` after transaction is created.
+
+:::tip
+After calling `transactionInitialize` or `transactionProcess` you can determine if `Checkout` has been completed by checking the `TransactionEvent` object returned in the mutation response.
+
+Transaction is considered "paid" when it has one of the following `types`:
```

Comment:
"it" was referring to `TransactionEvent`, but you are right that this depends on a context.

The reason I have this tip in the first place, is to show a way of determining if a Checkout could have been completed and how to determine if a `TransactionItem` was considered when Saleor calculated `Checkout.authorizeStatus` based on a response from `transactionInitialize` or `transactionProcess` mutation.

Storefront needs some way to know if it can show "success" screen / redirect to order details page.

Same thing applies to this comemnt: https://github.com/saleor/saleor-docs/pull/1365/files#r1806566297

Reply:
> Checkout could have been completed
> Storefront needs some way to know if it can show "success" screen / redirect to order details page.

This is why we have `checkout.authorizeStatus`, you also receive the `transactionEvent` as a mutation response, so you can use it to validate the current status. 

What is the use-case that would require storefront to determine if checkout can be closed by calculating the amounts on `checkout.transactions` that would not be handled by checkout.authorizeStatus? 🤔